File: Fundamentals/Dictionaries - Exercise/company_users.py
commands = input()
company = dict()
while commands != "End":
    commands = commands.split(" -> ")
    company_name = commands[0]
    employee_id = commands[1]
    if company_name not in company.keys():
        company[company_name] = list()
    if employee_id not in company[company_name]:
        company[company_name].append(employee_id)

    commands = input()
for company_name, employees in company.items():
    print(f"{company_name}")
    for employee in employees:
        print(f"-- {employee}")

File: JavaFundamentals/src/MapsLambdaAndStreamAPI/Exercise/CompanyUsers.java
package MapsLambdaAndStreamAPI.Exercise;

import java.util.*;

public class CompanyUsers {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        Map<String, Set<String>> companyUsers = new LinkedHashMap<>();

        String input;
        while (!(input = scanner.nextLine()).equals("End")) {
            String[] data = input.split(" -> ");

            String companyName = data[0];
            String employeeId = data[1];

            companyUsers.computeIfAbsent(companyName, k -> new LinkedHashSet<>()).add(employeeId);
        }

        for (Map.Entry<String, Set<String>> entry : companyUsers.entrySet()) {
            System.out.println(entry.getKey());
            for (String employeeId : entry.getValue()) {
                System.out.println("-- " + employeeId);
            }
        }
    }
}
